wagtail/admin/templates/wagtailadmin/home/locked_pages.html

```diff
@@ -31,7 +31,7 @@ <h2>{% trans "Your locked pages" %}</h2>
                                     {% if page.has_unpublished_changes %}
                                         <li><a href="{% url 'wagtailadmin_pages:view_draft' page.id %}" class="button button-small button-secondary" target="_blank" rel="noopener noreferrer">{% trans 'Draft' %}</a></li>
                                     {% endif %}
-                                    {% if page.live %}
+                                    {% if page.live_and_routable %}
```

Comment:
Because `live_and_routable` also checks the page's URL, this effectively makes two duplicate lookups to `page.url`, potentially hitting the database more than once each time. How would you feel about instead doing a single lookup, for example:

```html
{% if page.live %}
{% with page_url = page.url %}
    {% if page_url is not None %}
        <li><a href="{{ page_url }}" ...></li>
    {% endif %}
{% endwith %}
{% endif %}
```

Reply:
Nice, this is a better approach. I'll work on this, thanks!